.pre-commit-config.yaml

```diff
@@ -1,32 +1,46 @@
 repos:
-  # The warnings/errors we check for here are:
-  # E101 - mix of tabs and spaces
-  # W191 - use of tabs
-  # W291 - trailing whitespace
-  # W292 - no newline at end of file
-  # W293 - trailing whitespace
-  # W391 - blank line at end of file
-  # E111 - 4 spaces per indentation level
-  # E112 - 4 spaces per indentation level
-  # E113 - 4 spaces per indentation level
-  # E303 - too many blank lines (3)
-  # E304 - blank lines found after function decorator
-  # E305 - expected 2 blank lines after class or function definition
-  # E306 - expected 1 blank line before a nested definition
-  # E502 - the backslash is redundant between brackets
-  # E722 - do not use bare except
+# The warnings/errors we check for here are:
   # E901 - SyntaxError or IndentationError
   # E902 - IOError
-  # F822: undefined name in __all__
-  # F823: local variable name referenced before assignment
-  - repo: https://gitlab.com/pycqa/flake8
-    rev: 3.7.9
+  # F822 - undefined name in __all__
+  # F823 - local variable name referenced before assignment
+  # Others are taken care of by autopep8
+  - repo: https://github.com/PyCQA/flake8
+    rev: 4.0.1
     hooks:
       - id: flake8
-        args: ['--count', '--select', 'E101,W191,W291,W292,W293,W391,E111,E112,E113,E303,E304,E306,E502,E722,E901,E902,F822,F823']
```

Comment:
Why are we doing fewer checks here?  They seem like useful ones.

Reply:
less checks here but the other are fixed automatically by autopep8. I just copied and modified from sunpy

Reply:
I think autopep8 is pretty comprehensive have a look at the list of checks here https://pypi.org/project/autopep8/#features